src/unit_tests/os_auth/test_auth_validate.c

```diff
@@ -0,0 +1,344 @@
+    err = w_auth_validate_data(response,EXISTENT_IP1, NEW_AGENT1, NULL);  
+    assert_int_equal(err, OS_INVALID);
+    assert_string_equal(response, "ERROR: Duplicated IP: "EXISTENT_IP1"\n\n");  
+
+    /* Existent Agent Name */
+    response[0] = '\0'; 
+    expect_string(__wrap__merror, formatted_msg, "Invalid agent name "EXISTENT_AGENT1" (duplicated)");        
+    err = w_auth_validate_data(response,NEW_IP1, EXISTENT_AGENT1, NULL);  
+    assert_int_equal(err, OS_INVALID);
+    assert_string_equal(response, "ERROR: Duplicated agent name: "EXISTENT_AGENT1"\n\n");  
+   
+   /* Manager name */
+   char host_name[512];
+    if (gethostname(host_name, sizeof(shost) - 1) < 0) {
+        strncpy(host_name, "localhost", sizeof(host_name) - 1);
+        host_name[sizeof(host_name) - 1] = '\0';
+    }
+    char err_response[2048];    
+    snprintf(err_response, 2048, "ERROR: Invalid agent name: %s\n\n", host_name) ;
+    char merror_message[2048];    
+    snprintf(merror_message, 2048, "Invalid agent name %s (same as manager)", host_name);  
+    expect_string(__wrap__merror, formatted_msg, merror_message);        
+    err = w_auth_validate_data(response,NEW_IP1, host_name, NULL);
+    assert_int_equal(err, OS_INVALID);
+    assert_string_equal(response, err_response);  
+
+    /* Check no agent was deleted*/
+    assert_true(keys.keysize == 3);
+    int index = 0;
+    index = OS_IsAllowedName(&keys, EXISTENT_AGENT1);
+    assert_true(index >= 0);
+    index = OS_IsAllowedName(&keys, EXISTENT_AGENT2);
+    assert_true(index >= 0);
+}
+
+static void test_w_auth_validate_data_force_insert(void **state) {    
+
+    char response[2048] = {0};   
+    w_err_t err;
+
+    /* Duplicated IP*/
+    response[0] = '\0';
+    expect_string(__wrap__minfo, formatted_msg, "Duplicated IP '"EXISTENT_IP1"' (001). Saving backup.");            
+    err = w_auth_validate_data(response, EXISTENT_IP1, NEW_AGENT1, NULL);  
+    assert_int_equal(err, OS_SUCCESS);
+    assert_string_equal(response, "");  
+
+     /* Duplicated Name*/
+    response[0] = '\0';
+    expect_string(__wrap__minfo, formatted_msg, "Duplicated name '"EXISTENT_AGENT2"' (002). Saving backup.");
+    err = w_auth_validate_data(response, NEW_IP2, EXISTENT_AGENT2, NULL);  
+    assert_int_equal(err, OS_SUCCESS);
+    assert_string_equal(response, "");    
+
+    /* Check agents were deleted*/
+    int index = 0;
+    index = OS_IsAllowedIP(&keys, EXISTENT_IP1);
+    assert_true(index < 0);
+    index = OS_IsAllowedName(&keys, EXISTENT_AGENT2);
+    assert_true(index < 0);  
+}
+
+static void test_w_auth_validate_data_register_limit(void **state) {
+    char response[2048] = {0};   
+    char agent_name[2048];
+    char error_message[2048];
+    w_err_t err;
+
+    
+    //Filling most of keys element with a fixed key to reduce computing time
+    char fixed_key[KEYSIZE] = "1234";    
+    for(unsigned i=0; i<MAX_AGENTS-10; i++) {
+        OS_AddNewAgent(&keys, NULL, agent_name, ANY_IP, fixed_key);
```

Comment:
Valgrind: Conditional jump or move depends on uninitialised value(s)
agent_name has not been initialized when calling this function

Reply:
Will fix